css-scroll-snap-2/Overview.bs

```diff
@@ -275,17 +275,125 @@ Snap Events {#snap-events}
 	<table class="data" id="eventhandlers">
 		<thead>
 			<tr>
-				<th>Event handler
-				<th>Event handler event type
+				<th>Event</th>
+				<th>Interface</th>
+				<th>Targets</th>
+				<th>Description</th>
+			</tr>
+		</thead>
 		<tbody>
 			<tr>
-				<th><dfn event>snapChanged</dfn>
-				<td>{{scroll!!event}}
+				<th><dfn for="snapchanged" event>snapchanged</dfn></th>
+				<td>{{SnapEvent}}</td>
+				<td>scroll containers</td>
+				<td>Fired at the scrolling element or {{Document}} at the end of a scroll (before a {{scrollend}} event)
+					or after a <a href="https://drafts.csswg.org/css-scroll-snap-1/#re-snap">layout snap</a>
+					if the element that the scrolling element or Document is snapped to changed.</td>
+			</tr>
 			<tr>
-				<th><dfn event>snapChanging</dfn>
-				<td>{{scroll!!event}}
+				<th><dfn for="snapchanging" event>snapchanging</dfn></th>
+				<td>{{SnapEvent}}</td>
+				<td>snap containers</td>
+				<td>Fired at the scrolling element or {{Document}} during scrolling (similar to {{scroll}} events),
+					(before a {{snapchanged}} event, if one fires) if the scrolling will lead to a change in the
+					element that the scrolling element or Document is snapped to.</td>
+			</tr>
+		</tbody>
 	</table>
 
+	<h4 for="snapchanged" id="snapchanged"> snapchanged </h4>
+	{{snapchanged}} indicates that the snap area to which a snap container is snapped along either axis has changed.
+	{{snapchanged}} should be dispatched:
+
+	<ol>
+	<li>
+		at the end of a scrolling operation, after the user agent has completed its
+		snapping, if either:
+		<ul>
+		<li>	the snap area which the snap container is snapped to (TODO: link spec on how the target is picked) in the block axis is different
+			from the snap area it most recently snapped to in the block axis, or
+		</li>
+		<li>	the snap area it is snapped to in the inline axis is different
+			from the snap area it most recently snapped to in the inline axis.
+		</li>
+		</ul>
+	</li>
+	<li> (for snap containers with ''scroll-snap-type/proximity'' strictness) at the end of a scroll in which
+		the snap container scrolls and settles into (or out of) a
+		snap area's range of proximity snap positions in either axis.
+	</li>
+	<li> if there is a change to a snap container's style such that it goes from having a
+		non-'none' value for scroll-snap-type to having a 'none' value or vice versa.
```

Comment:
This should have well specified timing, maybe this occurs as part of "run the scroll steps" as well? https://drafts.csswg.org/cssom-view/#document-run-the-scroll-steps . I'm not sure if we would expect snap events for a non snapping scroller.

Reply:
I've added an "Appendix B" section as well as changes to the [Events](https://drafts.csswg.org/cssom-view/#events) section of cssom-view-1 which integrates with the [scroll steps](https://drafts.csswg.org/cssom-view/#document-run-the-scroll-steps).